### DISCUSSION TITLE
[Sidekick| Creating Components from Indexes not possible

### DISCUSSION THREAD

**Discussion by clark-ja** [Vector35/binaryninja-api#5857] [General]
Hello,

as the title says I cannot find any way to create components from Indexes as described in the [Sidekick documentation](https://docs.sidekick.binary.ninja/guide/components/#how-to-use-it)

> Components can also be created for the set of functions within the current index of the Sidekick Indexes sidebar. To create components for the functions in the current index, click on the Create Components in Current Index item in the Plugins/Sidekick menu or Create components for functions in this index icon in the Sidekick Indexes sidebar

However with the current stable binja and sidekick versions these options do not show up for me.

```
Binary Ninja Version: Personal 4.1.5902 Stable
Sidekick Version: 2.0.176
```

Is there something I am misunderstanding or is the documentation outdated and this is no longer possible?

Thanks for your help!

**Comment by kristopax** [comment]
Hi,

The documentation is outdated. The feature to use an action to create a component from functions in the current index is supported in Sidekick 1.x. To achieve the same effect in Sidekick 2.0, you can use the Analysis Workbench to write a script that opens a given index, generates a set of functions from the index entries, and then creates a component for each of those functions.

We will update the documentation. Thanks of reporting this.

**Comment by kristopax** [comment]
The documentation has been updated to indicate that a version of Sidekick before 2.0 is required to perform the operation of creating components for functions in the current index.
